autoPyTorch/api/tabular_classification.py

```diff
@@ -204,31 +263,17 @@ def search(
             self
 
         """
-        if dataset_name is None:
-            dataset_name = str(uuid.uuid1(clock_seq=os.getpid()))
-
-        # we have to create a logger for at this point for the validator
-        self._logger = self._get_logger(dataset_name)
-
-        # Create a validator object to make sure that the data provided by
-        # the user matches the autopytorch requirements
-        self.InputValidator = TabularInputValidator(
-            is_classification=True,
-            logger_port=self._logger_port,
-        )
 
-        # Fit a input validator to check the provided data
-        # Also, an encoder is fit to both train and test data,
-        # to prevent unseen categories during inference
-        self.InputValidator.fit(X_train=X_train, y_train=y_train, X_test=X_test, y_test=y_test)
+        assert isinstance(self.resampling_strategy, (CrossValTypes, HoldoutValTypes)), \
+            "Val Split is required for HPO search. " \
+            "Expected 'self.resampling_strategy' in" \
+            " '(CrossValTypes, HoldoutValTypes) got {}".format(self.resampling_strategy)
```

Comment:
Do we need `NoResamplingStrategyTypes` as well?

```suggestion
        if not isinstance(self.resampling_strategy, (CrossValTypes, HoldoutValTypes)):
            raise ValueError(
                "resampling_strategy (or validation split) is required for hyperparameter optimization."
                "Expected CrossValTypes or HoldoutValTypes, but got {}".format(self.resampling_strategy)
            )
```